File: src/FitMaster.Domain/Entities/Memberships/Package.cs
using FitMaster.Domain.Common;
using FitMaster.Domain.Entities.Payments;
using FitMaster.Domain.Enums;

namespace FitMaster.Domain.Entities.Memberships;


public class Package : BaseEntity
{
    public required string Name { get; set; }

    public string? Description { get; set; }

    public required decimal Price { get; set; }

    public required int DurationInDays { get; set; }

    public PackageStatus Status { get; set; } = PackageStatus.Active;

    public bool Deleted { get; set; }

    public ICollection<Membership> Memberships { get; set; } = new List<Membership>();

    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
}

File: frontend/src/types/package.ts
export type PackageStatus = "ACTIVE" | "INACTIVE";

export interface Package {
  id: string;
  name: string;
  price: number;
  durationInDays: number;
  status: PackageStatus | string;
  description: string;
}

export interface CreatePackageRequest {
  name: string;
  price: number;
  durationInDays: number;
  description: string;
  status: PackageStatus | string;
}

// PUT /api/packages/{id} requires the full object, not a partial patch.
export interface UpdatePackageRequest {
  id?: number;
  name: string;
  price: number;
  durationInDays: number;
  description: string;
  status: PackageStatus | string;
}
